File: Campus_Services_Portal/Models/Entities/Hostel.cs
namespace Campus_Services_Portal.Models.Entities
{
    public class Hostel
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Location { get; set; } = string.Empty;

        // One Hostel can have many Rooms
        public ICollection<Room> Rooms { get; set; } = new List<Room>();

        // One Hostel can have many Applications
        public ICollection<HostelApplication> HostelApplications { get; set; }
            = new List<HostelApplication>();
    }
}

File: campusx-frontend/src/app/core/models/hostel.ts
export interface Hostel {
  id: number;
  name: string;
  location: string;
  rooms: Room[];
}

export interface Room {
  id: number;
  roomNumber: string;
  capacity: number;
}

export interface HostelApplication {
  id: number;
  hostelName: string;
  preferences: string;
  status: string;
  roomNumber: string | null;
  appliedAt: string;
  updatedAt: string | null;
}